packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
I don't know how to make it meet @SeanTAllen's criteria of making it compatible with `String.add` for `ByteSeq` (or even if that is possible), but here is one implementation that works for my criteria.  Although it's probably not the most efficient way to do it, and could be improved, it illustrate that the caps can work (for my criteria at least):

```pony
  fun add(that: this->Array[A]): this->Array[this->A!] iso^ =>
    """
    Return a new Array that is a concatenation of this and that.
    """
    let len = _size + that._size
    var a = recover iso Array[this->A!](len) end
    
    for this_item in this.values() do a.push(this_item) end
    for that_item in that.values() do a.push(that_item) end
    
    consume a
```
```pony
actor Main
  new create(env: Env) =>
    let a: Array[U8] val = recover val [as U8: 1, 2, 3] end
    let b: Array[U8] val = recover val [as U8: 1, 2, 3] end
    let c: Array[U8] val = a + b
```